### PR TITLE
First pass at an Elixir style guide

### PR DESCRIPTION
**Pull request by BlakeWilliams**
The sample is a little lacking since I didn't have much time to flesh it
out. Would love ideas for good example functions.

### CODE REVIEW

File: style/elixir/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Elixir
+======
+
+[Sample](sample.ex)
+
+* Avoid macros.
```
Comment (drapergeek) [inline style/elixir/README.md]: Is this a style choice?

File: style/elixir/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+* Don't separate multiple declarations of the same function with a newline.
```
Comment (drapergeek) [inline style/elixir/README.md]: Is this a common thing in the community? I haven't followed this pattern so I'm hesitant on giving it a thumb without seeing it in action.

File: style/elixir/README.md (hunk 1 above)
Comment (BlakeWilliams) [inline style/elixir/README.md]: It is not, this probably belongs in best practices. I'll move it there.

File: style/elixir/README.md (hunk 2 above)
Comment (BlakeWilliams) [inline style/elixir/README.md]: I'm not sure to be honest. It's something I've done and I threw it in here to get some feedback on it.

**Comment by derekprior** [comment]
Is there a community style guide with any momentum we can adopt instead?

**Comment by BlakeWilliams** [comment]
Looked it up and there's https://github.com/niftyn8/elixir_style_guide#source-code-layout which seems to be relatively consistent with what we do.

**Comment by derekprior** [comment]
Reviewing that, here's what I'd take issue with:

> Use `do:` for single line if/unless statements.

I'd still rather see indention as it signals complexity that was introduced by the if to me. The one-liner hides that more than I'd like.

Everything else seemed fine to me. Not always what I'd do, but I'd be happy to abide by it.

**Comment by gylaz** [comment]
Idea: Could we do something like link to the community guide, then have a section for our *Deviations*?

**Comment by derekprior** [comment]
Yes. But I'm also willing to say "No deviations" and then debate each of them as they come up.

File: style/elixir/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+* Indent pipes by two spaces when using the match operator (`=`).
+* Don't indent pipes when not using the match operator (`=`).
```
Comment (gfontenot) [inline style/elixir/README.md]: I can't stand this practice. I don't understand why it's a thing. IMO, we should prefer consistency over (uninforced) semantic whitespace and just indent pipes 2 spaces all the time.

Comment (drapergeek) [inline style/elixir/README.md]: I agree with the consistency part but I'd go the other way and say don't indent ever.

**Comment by BlakeWilliams** [comment]
Closing because this is stale. Feel free to re-open if there's any interest.

**Comment by BlakeWilliams** [comment]
Reopening because of #460. I'd love feedback or additions here.

**Comment by genericlady** [comment]
@BlakeWilliams I went ahead and added to your branch. Should I push up changes or add recommendations to this thread for review?

**Comment by BlakeWilliams** [comment]
I think making a PR against this PR would work well, or adding the suggestions in text here. A PR might be easier for discussion though.

**Comment by drapergeek** [comment]
I think this looks great! How about we start here and we can all PR against your base file here? I have a few more I'd like to add.

**Comment by genericlady** [comment]
Started a modifications branch
[PR 482 modifications to bmw-elixir](https://github.com/thoughtbot/guides/pull/482)

**Comment by BlakeWilliams** [comment]
@drapergeek Sounds good, I'll go ahead and merge this one then.

**Pull request merged**